docs: initial implementation of error guides (#40060)
angular/angular: aio/content/errors/NG2003.md

@name Missing Token
@category compiler
@shortDescription No suitable injection token for parameter

@description

> There is no injection token for a class parameter at compile time. [InjectionTokens](https://angular.io/api/core/InjectionToken) are tokens that can be used in a Dependency Injection Provider.

Reviewer: ```suggestion
There is no injection token for a constructor parameter at compile time. [InjectionTokens](https://angular.io/api/core/InjectionToken) are tokens that can be used in a Dependency Injection Provider.
```

@debugging

> Look at the parameter that throws the error and all uses of the class. This error is commonly thrown in the constructor of a class, for primitive types like `string`, `number`, `boolean`, and `Object`.

Reviewer: ```suggestion
Look at the parameter that throws the error and all uses of the class. This error is commonly thrown when a constructor defines parameters with primitive types like `string`, `number`, `boolean`, and `Object`.
```
I started making a suggestion to drop the comma after "class" but ended up rewording a bit

> Use the [@Injectable](https://angular.io/api/core/Injectable) method or [@Inject](https://angular.io/api/core/Inject) decorator from `@angular/core` to ensure the type you are injecting is reified (has a runtime representation). Make sure to add a provider to this decorator so that you do not throw a [common runtime error](https://angular.io/error/NG0201).

Author: This link is contained in this PR so it will only work once merged. This links to the guide for NG0201 - I'll change the link to contain the page title as suggested in another comment for accessibility.
